### DISCUSSION TITLE
Can I use a distribution as input?

### DISCUSSION THREAD

**Discussion by tracycuixin** [StatMixedML/XGBoostLSS#29] [Q&A]
From my understanding, the model is taking one value with serval noise as input, and output a distribution.
Is it possible that I take a distribution as input and also output a distribution as the prediction?

**Comment by StatMixedML** [comment]
Dear @tracycuixin,

thanks for your interest in the project. I am not sure I properly understand your question. The model takes a response variable $\boldsymbol{y} = [y_{1}, y_{2}, y_{3}, \ldots, y_{N}]$ as input, where $N$ denotes your sample size. Given the specified distribution it predicts the distributional parameters. For a Gaussian, these would be $[\mu_{i}, \sigma_{i}], i=1, \ldots, N$.

What do you mean with taking a distribution as an input?

**Reply by tracycuixin** [reply]
Hi StatMixedML,
thanks for the response.

To confirm I understand you right, and please correct me if I am wrong. Here is my understanding

For example in the [simulation_example_Gaussian.ipynb]:
this shows the training data. And the y is the dependent variable, the x is the independent variable. (x1,x2....x10 are just the noise variables)
![image](https://user-images.githubusercontent.com/57021555/219822702-70dcd75b-d572-40f8-b509-e373798034df.png)

And this shows the the predicted distribution accordingly. Here the location and the scale are the mean and standard deviation of the predicted distribution.
![image](https://user-images.githubusercontent.com/57021555/219822863-e54183a8-8676-4a92-b656-9e8187e4e98e.png)

I plot one of the predicted distribution too
![image](https://user-images.githubusercontent.com/57021555/219823070-87d96a4f-2dfb-4ecb-86c8-81ad900c07bc.png)

I think my questions are:
1. if we use the model, do we also have to input the noise data too.
2. if we use the model, can we use the dataset that has more than one features
3. Can our feature also be a distribution? As in can we use distributions or distributional parameters as feature, so that this would be a distribution to distribution regression.

**Comment by StatMixedML** [comment]
@tracycuixin Thanks for your detailed explanations.

I haven't come across distribution to distribution regression so far. Is it related to Functional Data Analysis? Can you recommend some paper I can start with? Concerning your questions:

1. The noise variables were included to showcase that the model selects the most informative feature, for each distributional parameter individually. So you can include any feature you consider important for your project.
2. When you refer to "feature" you probably mean x-features? If so, yes you can include as many x-features you consider important for your project. 
3. I am not too familiar with distribution to distribution regression. But x can in principle be any feature.
